### PR TITLE
mdInkRipple disabled fix

### PR DESCRIPTION
**Pull request by bolerodan**
So while the previous PR for issue #344 silences the undefined error that was being thrown in getClosestPositionParent, it does not fix the issue at hand.

If disabled is set to true, the watcher calls this.destroy() removing the rippleElement from the parent. When disabled is set to false, the watcher calls this.init(). However at that point, rippleElement is no longer in the DOM, thus causing getClosestPositionedParent to throw the error originally.

The new fix currently just silences it, and currently in develop, does not re-append the rippleElement so when you toggle between disabled true/false, the rippleElement is never re-activated.

This checks in the watcher, and re-appends the rippleElement, however there are edge-cases.

If the button starts disabled, references to the parentElement are not caught because they are called in init(), and mounted checks for disabled, thus a null would be thrown in the watcher. I had to remove those out of init and set them in mounted so references are always stored for edge cases.

However another problem arises in the template. If disabled is set to true to begin with, mounted does not have the original DOM element, thus causing issues with stored references, so I removed the v-if from the template. I'm not 100% sure why that was there originally.

Take a look at this. I hope this brings up the original issue at hand and you can choose how to fix it.

Thanks a lot for you work.

### CODE REVIEW

**Comment by marcosmoura** [comment]
This was already fixed on v0.7.0. If something appear back I will take a deep look at this solution. Thanks a lot for this PR! :D

**Comment by bolerodan** [comment]
@marcosmoura  Are you sure? The code that this PR is against, hasnt changed between my PR and the release of 0.7.0

I will put up a fiddle soon. All you need to do, is toggle the disable for an md-button (or md-ink actually). So disable it, then Re-enable it. you will notice that md-ink-ripple no longer works.

The original PR technically did not solve what was actually happening, it just covered up the error ;)

TL;DR if you disable a button, then re-enable it, md-ink-ripple no longer works (but doesnt crash)

**Comment by bolerodan** [comment]
@marcosmoura   Here is a pen example

http://codepen.io/anon/pen/KaGpLg?editors=1111

click button. Ripple works. Click the checkbox to disable. Uncheck the checkbox to re-enable. Click button, ripple no longer works.

**Comment by terpret** [comment]
Same here.
